critiquebrainz/frontend/forms/review.py

```diff
@@ -16,7 +16,15 @@ def __call__(self, form, field):
         if form.state.data == "draft":
             return
         l = len(field.data) if field.data else 0
-        if l < self.min or self.max != -1 and l > self.max:
+        if (l < self.min or self.max != -1 and l > self.max) and (l != 0):
+            raise ValidationError(self.message)
+
+class TextAndRatingBothNotEmpty(object):
+    def __init__(self, message):
+        self.message = message
+
+    def __call__(self, form, field):
+        if not field.data and not form.rating.data:
```

Comment:
Accessing other fields in a validator doesn't seem right. `Form` class uses a `validate` method which you can override and check that either text or rating are present. Make sure to call original method with `super().validate()`.